### PR TITLE
Feature / 15 optimize memory repositories

### PR DESCRIPTION
**Pull request by axelsheva**

### CODE REVIEW

File: core/repository/block/headers.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { BlockId } from 'ddk.registry/dist/model/common/type';
+
+import { BlockHeaders } from 'shared/model/headers/block';
+
+export const BlockHeadersRepository = new Map<BlockId, BlockHeaders>();
```
Comment (macaqueee) [inline core/repository/block/headers.ts]: this is not repository :)

Comment (Verbalist) [reply core/repository/block/headers.ts]: BlockHeadersStorage

File: core/service/blockStorage.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        if (BlockRepository.size <= MEMORY_BLOCKS_LIMIT) {
+            return;
+        }
+
+        const removedBlock = BlockRepository.deleteFirst();
+        if (!removedBlock) {
+            return;
+        }
+
+        removedBlock.transactions.forEach(transaction => TransactionRepository.delete(transaction.id));
```
Comment (macaqueee) [inline core/service/blockStorage.ts]: lines from 21 to 26 should be replaced with some cleanup method.

File: core/service/blockStorage.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        return BlockPGRepository.getById(id);
+    }
+
+    getHeaders(id: BlockId): BlockHeaders {
+        return BlockHeadersRepository.get(id);
+    }
+
+    async getMany(limit: number, offset: number = 0): Promise<ResponseEntity<Array<Block>>> {
```
Comment (macaqueee) [inline core/service/blockStorage.ts]: need to think about Type of returning value. Maybe we don`t need ResponseEntity in Repository level anymore

File: core/service/round.ts [hunk 4]
```diff
@@ -158,7 +159,7 @@ class RoundService implements IRoundService {
     }
 
     public generate(firstSlotNumber: number): Round {
-        const lastBlockId = FailService.getRightLastRoundBlockId(BlockRepository.getLastBlock().id);
+        const lastBlockId = FailService.getRightLastRoundBlockId(BlockStorageService.getLast().id);
```
Comment (macaqueee) [inline core/service/round.ts]: extract to local variable `lastBlock`

File: core/repository/block/headers.ts (hunk 1 above)
Comment (axelsheva) [reply core/repository/block/headers.ts]: Why not? It produces methods for setting, getting, deleting, etc.

Comment (macaqueee) [reply core/repository/block/headers.ts]: What if you need custom logic. For example, `give me headers with those Ids`

File: core/service/blockStorage.ts (hunk 3 above)
Comment (Verbalist) [reply core/service/blockStorage.ts]: We can have problem from db level

File: core/repository/block/headers.ts (hunk 1 above)
Comment (axelsheva) [reply core/repository/block/headers.ts]: When I will create a class with this logic, but now this is not necessary

**Pull request merged**